src/averageAges.js

```diff
@@ -55,7 +69,36 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mens = filterByGender(people, 'm');
+  const allChild = !onlyWithSon
+    ? people.filter(child => findMother(people, child))
+    : mens.filter(child => findMother(people, child));
```

Comment:
we can optimize it. there is no need to filter men if `onlyWithSon `argument isn't passed. So better use this query for creating sample(men or all people) where we will `findMother`.